### ISSUE TITLE
Refine entity-aware boosting: dynamic weights based on question context

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#571]
## Problem

Current entity boosting uses fixed multipliers:
- PERSON: 2.5x
- PLACE: 2.5x
- PLACE + PERSON: 3.75x

These weights are static and don't consider:
- Question type ("Kiu" should boost PERSON more than "Kie")
- Entity co-occurrence patterns (Zamenhof + Esperanto frequently together)
- Document quality signals

## Solution

Dynamic entity boosting based on question analysis and entity relationships.

**Enhanced algorithm:**
```python
def compute_entity_boost(doc, query_analysis):
    boost = 1.0
    
    # 1. Question type matching
    if query_analysis['type'] == 'WHO':
        boost *= 3.0 if has_person_entity(doc) else 1.0
    elif query_analysis['type'] == 'WHERE':
        boost *= 3.0 if has_place_entity(doc) else 1.0
    elif query_analysis['type'] == 'WHEN':
        boost *= 2.5 if has_time_entity(doc) else 1.0
    
    # 2. Query entity co-occurrence
    query_entities = extract_entities(query_analysis)
    doc_entities = extract_entities(doc)
    
    if len(set(query_entities) & set(doc_entities)) >= 2:
        boost *= 1.5  # Both entities mentioned together
    
    # 3. Entity type diversity
    entity_types = get_entity_types(doc)
    if len(entity_types) >= 2:
        boost *= 1.2  # Rich entity context
    
    return boost
```

**Training data for statistics:**
- Mine entity co-occurrence patterns from corpus
- Analyze evaluation results: which boosting patterns worked?

## Expected Impact

+5-10% ranking accuracy by better matching documents to question intent.

## Dependencies

- Benefits from #552 (entity type classification) for more accurate type detection
- Independent otherwise

## Implementation

1. Analyze current boosting effectiveness on evaluation results
2. Design dynamic boosting formula based on patterns
3. Update `ast_aware_retriever.py` boosting logic
4. Mine entity co-occurrence statistics from corpus
5. Re-evaluate on test set

**Estimated time:** 2-3 days
**Params:** 0 (deterministic rules + corpus statistics)

**Comment by marctjones** [comment]
Closing as part of a 2026-09-08 backlog cleanup (see PLAN.md, DESIGN.md, CLAUDE.md's merge gate).

This issue predates the merge gate (adopted 2026-07-13) and the AST-native orchestrator rewrite. It describes work against the pre-orchestrator RAG pipeline or the learned model stack (M1/M2 embeddings, compositional embeddings, entity classifiers, neural rerankers) that was pruned from the repo -- klareco/embeddings/, klareco/models/, and klareco/summarization/ no longer exist. It has no milestone and was never benchmarked, matching exactly the failure mode CLAUDE.md documents (25 capability issues filed 2026-05-26, 8 implemented, none benchmarked).

Labeled archived-2026-09 so this batch is recoverable: gh issue list --state closed --label archived-2026-09. If this capability is still wanted, re-file it against the current architecture naming the benchmark metric and test the merge gate requires.
